pkg/appregistry/appregistry.go

```diff
@@ -40,7 +42,17 @@ func (f *factory) New(options Options) (Client, error) {
 		return nil, err
 	}
 
-	transport := httptransport.New(u.Host, u.Path, []string{u.Scheme})
+	// A custom HTTPClient is used here since the default HTTPClients ProxyFromEnvironment
+	// uses a cache which won't let us update the proxy env vars.
+	cli := &http.Client{
+		Transport: &http.Transport{
+			Proxy: func(req *http.Request) (*url.URL, error) {
+				return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
```

Comment:
This will work if the proxy variables are not set in the environment.

Reply:
This does not return an error, we are setting the Proxy func: https://github.com/golang/go/blob/master/src/net/http/transport.go#L112-L121

Reply:
https://godoc.org/net/http#ProxyFromEnvironment

You could use that and make it more clear.

Also, this is how the default client works, so this might not even be needed

Reply:
Thanks @shawn-hurley, I tried using `ProxyFromEnvironment` but the result seems to be cached, setting the env variables did cause marketplace to use the proxy.